### PR TITLE
feat: position today in gantt

### PR DESCRIPTION
**Pull request by sherotree**
## What this PR does / why we need it:
1. make gantt to full screen
2. position today in gantt

## I have checked the following points:
- [x] I18n is finished and updated by cli
- [x] Form fields validation is added and length is limited
- [x] Display normally on small screen
- [x] Display normally when some data is empty or null
- [x] Display normally in english mode

## Does this PR introduce a user interface change?

✅ Yes(screenshot is required)

## ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English | Make Gantt chart have full screen and positioning to today function |
| 🇨🇳 中文    |  使甘特图有全屏和定位至今天功能   |

## Does this PR need be patched to older version?
✅ Yes(version is required)

## Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: shell/app/common/use-hooks.tsx [hunk 1]
```diff
@@ -872,8 +872,20 @@ type IUseFullScreen = (
   options?: Options,
 ) => [boolean, { exitFullscreen: () => void; toggleFullscreen: () => void; enterFullscreen: () => void }];
 
+/**
+ * @description Get the latest status to prevent problems caused by closures
+ * @param value
+ * @returns
+ */
+export const useLatest = <T,>(value: T) => {
```
Comment (McDaddy) [inline shell/app/common/use-hooks.tsx:880]: what does `<T,>` mean?

File: shell/app/config-page/components/gantt/components/gantt/gantt.tsx [hunk 2]
```diff
@@ -220,6 +224,11 @@ export const Gantt: React.FunctionComponent<GanttProps> = ({
     }
   }, [taskListRef, listCellWidth]);
 
+  useEffect(() => {
+    scrollToToday();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [h_number, dateSetup]);
```
Comment (daskyrk) [inline shell/app/config-page/components/gantt/components/gantt/gantt.tsx:230]: Will position today when  change window size which will cause h_number change?

File: shell/app/modules/project/pages/issue/gantt/index.tsx [hunk 3]
```diff
@@ -210,10 +216,21 @@ const IssuePlan = () => {
       chosenIssueType: undefined,
     });
   };
+
+  const handleScreenChange = (value: boolean) => {
+    updater.isFullScreen(value);
+  };
+
+  const buttonEle = document.getElementsByClassName('top-button-group');
```
Comment (McDaddy) [inline shell/app/modules/project/pages/issue/gantt/index.tsx:224]: how about put these code under a useEffect hook.
when unmount make sure the style is restored.

File: shell/app/config-page/components/gantt/types/public-types.ts [hunk 4]
```diff
@@ -71,6 +75,11 @@ export interface EventOption {
    * Invokes on expander on task list
    */
   onExpanderClick?: (task: Task) => void;
+
+  /**
+   * Invokes on enter full screen or exit full screen
+   */
+  screenChange: () => void;
```
Comment (McDaddy) [inline shell/app/config-page/components/gantt/types/public-types.ts:82]: keep same name style with others, `onScreenChange`

File: shell/app/config-page/components/gantt/components/gantt/gantt.tsx [hunk 5]
```diff
@@ -224,10 +224,10 @@ export const Gantt: React.FunctionComponent<GanttProps> = ({
     }
   }, [taskListRef, listCellWidth]);
 
-  useEffect(() => {
+  useLayoutEffect(() => {
     scrollToToday();
     // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (McDaddy) [inline shell/app/config-page/components/gantt/components/gantt/gantt.tsx:229]: can this disable lint be removed?

File: shell/app/common/use-hooks.tsx [hunk 6]
```diff
@@ -874,6 +874,11 @@ type IUseFullScreen = (
 
 export const useFullScreen: IUseFullScreen = (target, options) => {
   const [state, setState] = React.useState(false);
+  const latestStateRef = React.useRef(false);
+
+  React.useEffect(() => {
+    latestStateRef.current = state;
+  }, [state]);
```
Comment (daskyrk) [inline shell/app/common/use-hooks.tsx:881]: Why not set `latestStateRef.current = state;` without hook.

**Pull request merged**